### PR TITLE
[cppkafka] Fix dynamic build.

### PR DESCRIPTION
**Pull request by NancyLi1013**
This PR will fix dynamic issue. 
Related issue #8067.

### CODE REVIEW

**Review by PhoebeHui — APPROVED** [review]

File: ports/cppkafka/portfile.cmake [hunk 1]
```diff
@@ -6,18 +6,36 @@ vcpkg_from_github(
...
     SHA512 60d01ce1dd9bd9119676be939ed5ab03539abb1f945c1b31e432edfe0f06542778f7fef37696f5ff19c53024f44d5cbd8aeddbbb231c38b098e05285d3ff0cab
     HEAD_REF master
+    PATCHES fix-dynamic.patch
 )
 
+string(COMPARE EQUAL ${VCPKG_LIBRARY_LINKAGE} "static" IS_STATIC)
+
+if (IS_STATIC)
+    set(CPPKAFKA_BUILD_SHARED OFF)
+    set(CPPKAFKA_BOOST_STATIC_LIBS ON)
```
Comment (Neumann-A) [inline ports/cppkafka/portfile.cmake]: that is probably incorrect in custom triplets building this port statically and boost dynamically or vice versa. Is setting CPPKAFKA_BOOST_STATIC_LIBS really required or can it just find any available boost version without it?

Comment (cbezault) [inline ports/cppkafka/portfile.cmake]: You don't need to set `CPPKAFKA_BOOST_*_LIBS` we only ever provide either static or dynamic boost so it doesn't have a choice and the variable is ignored.

@Neuman-A 's comment might still stand for `CPPKAFKA_RDKAFKA_STATIC_LIB`. If it actually changes something we need to come up with a way of figuring out how to set it without assuming it was built in the same was as cppkafka.

**Review by cbezault — CHANGES_REQUESTED** [review]
LGTM minus the one comment.

File: ports/cppkafka/portfile.cmake (hunk 1 above)
Comment (NancyLi1013) [reply ports/cppkafka/portfile.cmake]: Thanks for your review and comments. I have checked the source code. `CPPKAFKA_RDKAFKA_STATIC_LIB` is used to set librdkafka prefix and suffix to judge the port is static or dynamic. So I think we should set `CPPKAFKA_RDKAFKA_STATIC_LIB` here.

Comment (Neumann-A) [reply ports/cppkafka/portfile.cmake]: Then just append the static prefix and suffixes to CMAKE_FIND_LIBRARY_(PREFIXES|SUFFIXES) (if they are not the defaults cmake already adds). Then CMake should be able to find the library correctly.

Comment (NancyLi1013) [reply ports/cppkafka/portfile.cmake]: @Neumann-A  Thanks again for your comments. I noticed that CMAKE_FIND_LIBRARY_SUFFIXES default has been overrided in `/cmake/FindRdkafka.cmake`.

```
# Override default CMAKE_FIND_LIBRARY_SUFFIXES
# (Allows optional prioritization of static libraries during resolution)
if (CPPKAFKA_RDKAFKA_STATIC_LIB)
    set(RDKAFKA_PREFIX ${CMAKE_STATIC_LIBRARY_PREFIX})
    set(RDKAFKA_SUFFIX ${CMAKE_STATIC_LIBRARY_SUFFIX})
else()
    set(RDKAFKA_PREFIX ${CMAKE_SHARED_LIBRARY_PREFIX})
    set(RDKAFKA_SUFFIX ${CMAKE_SHARED_LIBRARY_SUFFIX})
endif()
```

Comment (cbezault) [reply ports/cppkafka/portfile.cmake]: Add a cmake wrapper to librdkafka which just adds the variables the module exports.

```
# This find module helps find the RdKafka module. It exports the following variables:
# - RdKafka_INCLUDE_DIR : The directory where rdkafka.h is located.
# - RdKafka_LIBNAME : The name of the library, i.e. librdkafka.a, librdkafka.so, etc.
# - RdKafka_LIBRARY_PATH : The full library path i.e. <path_to_binaries>/${RdKafka_LIBNAME}
# - RdKafka::rdkafka : Imported library containing all above properties set.
```

**Comment by ray42** [comment]
I have been having this problem too. Is this ready to be merged in? Thanks!

**Comment by cbezault** [comment]
A cmake wrapper needs to be added to rdkafka

**Comment by Rastaban** [comment]
@NancyLi1013 can you add a vcpkg cmake wrapper to cppkafka?  If you search for "vcpkg-cmake-wrapper.cmake" under the ports directory then you should be able to find some good examples.

**Comment by PhoebeHui** [comment]
Thank you all for your guideline! I will help follow up.

**Pull request merged**